firedrake/bcs.py

```diff
@@ -295,3 +297,142 @@ def homogenize(bc):
     except TypeError:
         # not iterable
         return DirichletBC(bc.function_space(), 0, bc.sub_domain)
+
+
+class FormBC(object):
```

Comment:
Don't copy and paste code. Instead inherit either from DirichletBC or create a common ancestor.